### PR TITLE
Include Unchained Derivation Patterns

### LINKED ISSUE #1467: Model T no longer allows signing for arbitrary deriviation paths below `m/45'`
**Describe the bug**

Signing for arbitrary public keys with deriviation paths under `m/45'` used to work -- it no longer does.

**Firmware version and revision**
Model T 2.3.5

**Desktop/smartphone setup (please complete the following information):**
 - Wallet software: Unchained Capital platform & Caravan
 - OS: Seen on macOS but believed to be OS-independent issue
 - Browser: Seen on Chrome 87, but believed to be browser-independent issue

**To Reproduce**
1. Construct a transaction using public keys derived from a path such as `m/45'/0'/0'/0/0/0`(6 path components which, indeed is different than the BIP45 spec).
2. Sign the transaction with a pre-2.3.5 model T or with a Trezor ONE -- it works.
3. Sign the same transaction with a 2.3.5 model T, we get `Forbidden key path` error.

**Expected behavior**
We expected to be able to continue to sign for arbitrary derivation paths below `m/45'`.  This changed without any notification in advance, breaking signing for many of our customers.

**Screenshots**
Below is a screenshot of Caravan's [functional test suite](https://unchained-capital.github.io/caravan/#/test)
gscreenshot_2021-02-10-112401

Below are two screenshot of Unchained Capital's (test) application, first when signing for a transaction
gscreenshot_2021-02-10-114442

And then when confirming an address.
gscreenshot_2021-02-10-114557

### PR DESCRIPTION
**Pull request by bucko13**
Resolves #1467 

Our attempt to add the patterns we need to the firmware to restore prior functionality for our clients. Would love feedback or to see this merged.

Thanks!

### CODE REVIEW

**Comment by prusnak** [comment]
Range `[0-1000000]` seems a little bit excessive - are you really using that wide range?

**Comment by dhruvbansal** [comment]
> Range `[0-1000000]` seems a little bit excessive - are you really using that wide range?

In practice we do not saturate this range but we do have a lot of keys and addresses we've collaboratively built with our clients.  Specifying a lower upper limit feels like it publicly reveals too much about our business and creates an artificial ceiling that we may need to blow past one day depending on how our product roadmap evolves.

For these reasons, we'd actually prefer to be able to use `*` instead of specifying any range.  We went with `[0-1000000]` because it's the same range already being used for `address_index` -- https://github.com/trezor/trezor-firmware/blob/master/core/src/apps/common/paths.py#L100

I understand that limiting the acceptable ranges of path segments makes abuse of the paths more difficult.  But our application controls path usage and we ensure our clients have access to the information they need to verify paths on their devices, a crucial part of independent recovery.

**Comment by prusnak** [comment]
@matejcik can you please have a look and merge if the suggestion makes sense?

File: core/src/apps/bitcoin/keychain.py [hunk 1]
```diff
@@ -52,6 +52,9 @@ class MsgWithAddressScriptType(Protocol):
 
 PATTERN_CASA = "m/49/coin_type/account/change/address_index"
 
+PATTERN_UNCHAINED_HARDENED = "m/45'/coin_type'/account'/[0-1000000]/change/address_index"
+PATTERN_UNCAINED_UNHARDENED = "m/45'/coin_type/account/[0-1000000]/change/address_index"
```
Comment (matejcik) [inline core/src/apps/bitcoin/keychain.py:56]: typo

File: core/src/apps/bitcoin/keychain.py [hunk 2]
```diff
@@ -83,6 +86,9 @@ def validate_path_against_script_type(
         if coin.coin_name in BITCOIN_NAMES:
             patterns.append(PATTERN_GREENADDRESS_A)
             patterns.append(PATTERN_GREENADDRESS_B)
+            patterns.append(PATTERN_UNCHAINED_HARDENED)
```
Comment (matejcik) [inline core/src/apps/bitcoin/keychain.py:89]: just to confirm: are you sure you are only using **non-segwit** accounts? i.e., script type `SPENDADDRESS`/`SPENDMULTISIG` ?

Comment (bucko13) [reply core/src/apps/bitcoin/keychain.py:89]: Yes that's right. The recent changes only impact us for non-segwit based accounts.

**Review by prusnak — APPROVED** [review]

**Review by matejcik — APPROVED** [review]

**Comment by prusnak** [comment]
Superseded by https://github.com/trezor/trezor-firmware/pull/1473

Let's wait until that PR builds in the CI

**Pull request closed without merging**